File: scripts/i18n_audit.py
import os
import re
import json

ROOT = os.path.join(os.path.dirname(__file__), '..')
SRC = os.path.join(ROOT, 'frontend', 'src')
LOCAL = os.path.join(SRC, 'locales', 'zh-CN.json')

key_re = re.compile(r"t\(['\"]([^'\"]+)['\"]\)")

keys = set()
for dirpath, dirs, files in os.walk(SRC):
    for fname in files:
        if fname.endswith(('.js', '.jsx', '.ts', '.tsx')):
            fpath = os.path.join(dirpath, fname)
            try:
                with open(fpath, 'r', encoding='utf8') as f:
                    content = f.read()
            except Exception:
                continue
            for m in key_re.finditer(content):
                keys.add(m.group(1))

with open(LOCAL, 'r', encoding='utf8') as f:
    locales = json.load(f)

missing = sorted(k for k in keys if k not in locales)
unused = sorted(k for k in locales.keys() if k not in keys)

print(f'Found {len(keys)} unique t(...) keys in frontend source.')
print(f'Missing in zh-CN.json: {len(missing)}')
if missing:
    print('\n'.join(missing))
print('\nUnused keys in zh-CN.json:', len(unused))
if unused:
    print('\n'.join(unused))

File: scripts/i18n_audit.js
const fs = require('fs');
const path = require('path');

function walk(dir, ext, fileList = []) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const full = path.join(dir, file);
    const stat = fs.statSync(full);
    if (stat.isDirectory()) walk(full, ext, fileList);
    else if (ext.some(e => full.endsWith(e))) fileList.push(full);
  }
  return fileList;
}

const srcFiles = walk(path.join(__dirname, '..', 'frontend', 'src'), ['.js', '.jsx', '.ts', '.tsx']);
const keyRe = /t\(['\"]([^'\"]+)['\"]\)/g;
const keys = new Set();
for (const f of srcFiles) {
  const content = fs.readFileSync(f, 'utf8');
  let m;
  while ((m = keyRe.exec(content)) !== null) keys.add(m[1]);
}

const localesPath = path.join(__dirname, '..', 'frontend', 'src', 'locales', 'zh-CN.json');
const locales = JSON.parse(fs.readFileSync(localesPath, 'utf8'));

const missing = [];
for (const k of [...keys].sort()) {
  if (!(k in locales)) missing.push(k);
}

console.log('Found', keys.size, 'unique t(...) keys in frontend source.');
console.log('Missing in zh-CN.json:', missing.length);
if (missing.length) console.log(missing.join('\n'));

// Also check keys present in locales but not used
const unused = [];
for (const k of Object.keys(locales).sort()) {
  if (!keys.has(k)) unused.push(k);
}
console.log('\nUnused keys in zh-CN.json:', unused.length);
if (unused.length) console.log(unused.join('\n'));
